src/hardware/imf.cpp

```diff
@@ -0,0 +1,13422 @@
+template <typename... Args>
+void IMF_LOG(std::string format, Args const&... args)
+{
+	SDL_LockMutex(m_loggerMutex);
+	printf((format + "\n").c_str(), args...);
+	SDL_UnlockMutex(m_loggerMutex);
+}
+#else
+#	define IMF_LOG(...)
+#endif
+
+inline uint8_t leftRotate8(uint8_t n)
+{
+	return (n << 1) | (n >> (8 - 1));
+}
+
+enum DataParty : uint8_t {
+	DATAPARTY_SYSTEM,
+	DATAPARTY_MIDI,
+};
+
+static void Intel8253_TimerEvent(const uint32_t val);
+
+enum CARD_MODE : uint8_t {
+	MUSIC_MODE = 0,
+	THRU_MODE  = 1,
+};
+
+enum CHAIN_MODE : uint8_t {
+	CHAIN_MODE_DISABLED = 0,
+	CHAIN_MODE_ENABLED  = 1,
+};
+
+enum MEMORY_PROTECTION : uint8_t {
+	MEMORY_WRITABLE = 0,
+	MEMORY_READONLY = 1,
+};
+
+enum ERROR_REPORTING : uint8_t {
+	ERROR_REPORTING_DISABLED = 0,
+	ERROR_REPORTING_ENABLED  = 1,
+};
+enum HANDSHAKE_MESSAGE : uint8_t {
+	ACK_MESSAGE    = 2,
+	NAK_MESSAGE    = 3,
+	CANCEL_MESSAGE = 4,
+};
+
+enum MUSICCARD_ERROR_CODE : uint8_t {
+	FIFO_OVERFLOW_ERROR_MUSICCARD_TO_SYSTEM = 0xF0,
+	FIFO_OVERFLOW_ERROR_MIDI_TO_MUSICCARD   = 0xF1,
+	MIDI_RECEPTION_ERROR                    = 0xF2,
+	MIDI_OFFLINE_ERROR                      = 0xF3,
+	TIMEOUT_ERROR_MIDI_TO_MUSICCARD         = 0xF4,
+	TIMEOUT_ERROR_SYSTEM_TO_MUSICCARD       = 0xF5,
+};
+
+enum MidiDataPacketState : uint8_t {
+	PACKET_STATE_00 = 0x00,
+	PACKET_STATE_01 = 0x01,
+	PACKET_STATE_02 = 0x02,
+	PACKET_STATE_03 = 0x03,
+	PACKET_STATE_04 = 0x04,
+	PACKET_STATE_05 = 0x05,
+	PACKET_STATE_06 = 0x06,
+	PACKET_STATE_07 = 0x07,
+	PACKET_STATE_08 = 0x08,
+	PACKET_STATE_09 = 0x09,
+	PACKET_STATE_0A = 0x0A,
+	PACKET_STATE_0B = 0x0B,
+	PACKET_STATE_0C = 0x0C,
+	PACKET_STATE_0D = 0x0D,
+	PACKET_STATE_0E = 0x0E,
+	PACKET_STATE_0F = 0x0F,
+	PACKET_STATE_10 = 0x10,
+	PACKET_STATE_11 = 0x11,
+	PACKET_STATE_12 = 0x12,
+	PACKET_STATE_13 = 0x13,
+	PACKET_STATE_14 = 0x14,
+	PACKET_STATE_15 = 0x15,
+	PACKET_STATE_16 = 0x16,
+	PACKET_STATE_17 = 0x17,
+	PACKET_STATE_18 = 0x18,
+	PACKET_STATE_19 = 0x19,
+	PACKET_STATE_1A = 0x1A,
+	PACKET_STATE_1B = 0x1B,
+	PACKET_STATE_1C = 0x1C,
+	PACKET_STATE_1D = 0x1D,
+	PACKET_STATE_1E = 0x1E,
+	PACKET_STATE_1F = 0x1F,
+	PACKET_STATE_20 = 0x20,
+	PACKET_STATE_21 = 0x21,
+	PACKET_STATE_22 = 0x22,
+	PACKET_STATE_23 = 0x23,
+	PACKET_STATE_24 = 0x24,
+	PACKET_STATE_25 = 0x25,
+	PACKET_STATE_26 = 0x26,
+	PACKET_STATE_27 = 0x27,
+	PACKET_STATE_28 = 0x28,
+	PACKET_STATE_29 = 0x29,
+	PACKET_STATE_2A = 0x2A,
+	PACKET_STATE_2B = 0x2B,
+	PACKET_STATE_2C = 0x2C,
+	PACKET_STATE_2D = 0x2D,
+	PACKET_STATE_2E = 0x2E,
+	PACKET_STATE_2F = 0x2F,
+	PACKET_STATE_30 = 0x30,
+	PACKET_STATE_31 = 0x31,
+	PACKET_STATE_32 = 0x32,
+	PACKET_STATE_33 = 0x33,
+	PACKET_STATE_34 = 0x34,
+	PACKET_STATE_35 = 0x35,
+	PACKET_STATE_36 = 0x36,
+	PACKET_STATE_37 = 0x37,
+	PACKET_STATE_38 = 0x38,
+	PACKET_STATE_39 = 0x39,
+	PACKET_STATE_3A = 0x3A,
+	PACKET_STATE_3B = 0x3B,
+	PACKET_STATE_3C = 0x3C,
+	PACKET_STATE_3D = 0x3D,
+	PACKET_STATE_3E = 0x3E,
+	PACKET_STATE_3F = 0x3F,
+	PACKET_STATE_40 = 0x40,
+};
```

Comment:
All enum names and values should be UpperCamelCase (subject to much time you want to spend on this...)

Reply:
(marking resolved because not renaming variables)